frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceParentServlet.java

```diff
@@ -0,0 +1,89 @@
+package com.google.tripmeout.frontend.servlet;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import com.google.tripmeout.frontend.places.PlaceService;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PlaceParentServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final Pattern TRIP_NAME_PATTERN = Pattern.compile(".*/trips/([^/]+)/placeVisits");
+
+  private final PlaceVisitStorage placeStorage;
+  private final Gson gson;
+  private final PlaceService placeService;
+
+  public PlaceParentServlet(PlaceVisitStorage placeStorage, Gson gson, PlaceService placeService) {
+    this.placeStorage = placeStorage;
+    this.gson = gson;
+    this.placeService = placeService;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PrintWriter writer = response.getWriter();
+    try {
+      PlaceVisitModel place =
+          ServletUtil.extractFromRequestBody(request.getReader(), gson, PlaceVisitModel.class);
+
+      placeService.validatePlaceId(place.placesApiPlaceId());
+
+      String tripId = ServletUtil.matchUriOrThrowError(request, TRIP_NAME_PATTERN).group(1);
+      logger.atInfo().log("finished matching");
+
+      PlaceVisitModel newPlace = PlaceVisitModel.builder()
+                                     .setId(UUID.randomUUID().toString())
+                                     .setTripId(tripId)
+                                     .setPlaceName(place.placeName())
+                                     .setPlacesApiPlaceId(place.placesApiPlaceId())
+                                     .setUserMark(PlaceVisitModel.UserMark.YES)
+                                     .build();
+
+      PlaceVisitModel updatedPlace =
+          placeStorage.updateUserMarkOrAddPlaceVisit(newPlace, PlaceVisitModel.UserMark.YES);
+
+      response.setContentType("applciation/json");
+      writer.println(gson.toJson(updatedPlace));
+      response.setStatus(HttpServletResponse.SC_OK);
```

Comment:
Consider returning `SC_CREATED` (HTTP 201) here. If you do that, you may need to update client code. Either way, if we decide to return 201 for create, then all servlet posts should probably be updated (in which case we should add an issue to track the fix).